python/taichi/snode/fields_builder.py

```diff
@@ -139,3 +139,8 @@ def finalize(self):
     def _check_not_finalized(self):
         if self._finalized:
             raise InvalidOperationError('FieldsBuilder finalized')
+
+    def destroy(self):
```

Comment:
Consider adding this method to `SNodeTree` itself: The builder is for building the tree. What's the semantics of calling `destroy()` while it's not finalized?